DQMOffline/Trigger/plugins/ZGammaplusJetsMonitor.cc

```diff
@@ -0,0 +1,813 @@
+      FillME(a_ME_HE_I_highRefPt, DirectBalance_, DifJ1PtTrObjPt_, asymmetry, trigobj_pt[0], v_jetpt[0], true);
+    }
+  }  // end is endcap-inner
+  if (isEndCapOuter(v_jeteta[0])) {
+    FillME(a_ME_HE_O, DirectBalance_, DifJ1PtTrObjPt_, asymmetry, trigobj_pt[0], v_jetpt[0], true);
+    if (islowRefPt(trigobj_pt[0])) {
+      FillME(a_ME_HE_O_lowRefPt, DirectBalance_, DifJ1PtTrObjPt_, asymmetry, trigobj_pt[0], v_jetpt[0], true);
+    } else if (ismediumRefPt(trigobj_pt[0])) {
+      FillME(a_ME_HE_O_mediumRefPt, DirectBalance_, DifJ1PtTrObjPt_, asymmetry, trigobj_pt[0], v_jetpt[0], true);
+    } else if (ishighRefPt(trigobj_pt[0])) {
+      FillME(a_ME_HE_O_highRefPt, DirectBalance_, DifJ1PtTrObjPt_, asymmetry, trigobj_pt[0], v_jetpt[0], true);
+    }
+  }  // end is endcap-outer
+  if (isForward(v_jeteta[0])) {
+    FillME(a_ME_HF, DirectBalance_, DifJ1PtTrObjPt_, asymmetry, trigobj_pt[0], v_jetpt[0], true);
+    if (islowRefPt(trigobj_pt[0])) {
+      FillME(a_ME_HF_lowRefPt, DirectBalance_, DifJ1PtTrObjPt_, asymmetry, trigobj_pt[0], v_jetpt[0], true);
+    } else if (ismediumRefPt(trigobj_pt[0])) {
+      FillME(a_ME_HF_mediumRefPt, DirectBalance_, DifJ1PtTrObjPt_, asymmetry, trigobj_pt[0], v_jetpt[0], true);
+    } else if (ishighRefPt(trigobj_pt[0])) {
+      FillME(a_ME_HF_highRefPt, DirectBalance_, DifJ1PtTrObjPt_, asymmetry, trigobj_pt[0], v_jetpt[0], true);
+    }
+  }  // end is Forward
+}
+// ----- Reference'Pt categories: low[30,50), medium[50,100), high[100,inf) -----
+bool ZGammaplusJetsMonitor::islowRefPt(double refPt) {
+  bool output = false;
+  if (refPt >= 30. && refPt < 50.)
+    output = true;
+  return output;
+}
+
+bool ZGammaplusJetsMonitor::ismediumRefPt(double refPt) {
+  bool output = false;
+  if (refPt >= 50. && refPt < 100.)
+    output = true;
+  return output;
+}
+
+bool ZGammaplusJetsMonitor::ishighRefPt(double refPt) {
+  bool output = false;
+  if (refPt >= 100.)
+    output = true;
+  return output;
+}
+// --------- detector areas ---------------------------------------------------
+bool ZGammaplusJetsMonitor::isBarrel(double eta) {
+  bool output = false;
+  if (fabs(eta) <= 1.3)
+    output = true;
+  return output;
+}
+
+bool ZGammaplusJetsMonitor::isEndCapInner(double eta) {
+  bool output = false;
+  if (fabs(eta) <= 2.5 && fabs(eta) > 1.3)
+    output = true;
+  return output;
+}
+
+bool ZGammaplusJetsMonitor::isEndCapOuter(double eta) {
+  bool output = false;
+  if (fabs(eta) <= 3.0 && fabs(eta) > 2.5)
+    output = true;
+  return output;
+}
+
+bool ZGammaplusJetsMonitor::isForward(double eta) {
+  bool output = false;
+  if (fabs(eta) > 3.0)
+    output = true;
+  return output;
+}
+// --------- Matching ---------------------------------------------------------------
+bool ZGammaplusJetsMonitor::isMatched(double hltJetEta, double hltJetPhi, double OffJetEta, double OffJetPhi) {
+  bool output = false;
+  double DRMatched = 0.4;
+  double dR = deltaR(hltJetEta, hltJetPhi, OffJetEta, OffJetPhi);
```

Comment:
`deltaR2` is advantageous performance-wise.

Reply:
@esiam this comment was not addressed. Please use `deltaR2`

https://github.com/cms-sw/cmssw/blob/ec85b46f8bb344815571f6b9d1d46d54fd2ff1e0/DataFormats/Math/interface/deltaR.h#L16

which avoids a (costly) square root in tight loops.